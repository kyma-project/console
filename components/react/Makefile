ci-pr: resolve build validate test
ci-master: resolve build validate test
ci-release: resolve build validate test

resolve:
	npm install

build:
	npm run build

validate:
	npm run lint-check
	eslint -c .eslinterrc ./src

test:
	CI=true npm run test
	