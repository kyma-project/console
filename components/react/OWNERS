reviewers:
    - default-reviewers

approvers:
    - default-approvers
    - Wawrzyn321
    - content-reviewers