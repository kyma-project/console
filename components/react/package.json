{
  "name": "@kyma-project/react-components",
  "version": "0.0.43",
  "main": "dist/index.js",
  "private": false,
  "license": "Apache-2.0",
  "directories": {
    "dist": "src"
  },
  "scripts": {
    "start:kyma": "npm run build:watch",
    "start:kyma:api": "npm run build:watch",
    "build": "webpack --config webpack.config.prod.js",
    "build:watch": "webpack -w --config webpack.config.dev.js",
    "test": "jest src/**; exit 0",
    "test:watch": "jest src/** --watch",
    "test:watchAll": "jest src/** --watchAll",
    "test:coverage:report": "open ./coverage/lcov-report/index.html",
    "lint-fix": "prettier --config ../../.prettierrc --write \"./src/**/*.{js,jsx,ts,tsx,html,css,json}\"",
    "lint-check": "prettier --config ../../.prettierrc --check \"./src/**/*.{js,jsx,ts,tsx,html,css,json}\""
  },
  "dependencies": {
    "@babel/runtime-corejs2": "^7.12.5",
    "lodash": "^4.17.21",
    "polished": "^1.8.1",
    "styled-components-modifiers": "^1.0.1"
  },
  "devDependencies": {
    "@babel/types": "7.0.0-beta.44",
    "ajv": "^6.12.3",
    "babel-cli": "^6.26.0",
    "babel-core": "^6.26.0",
    "babel-eslint": "^8.0.1",
    "babel-jest": "^22.4.1",
    "babel-loader": "^7.1.2",
    "babel-polyfill": "^6.26.0",
    "babel-preset-env": "^1.6.1",
    "babel-preset-react": "^6.24.1",
    "babel-preset-stage-0": "^6.24.1",
    "enzyme": "^3.1.0",
    "enzyme-adapter-react-16": "^1.0.2",
    "exec-sh": "^0.3.3",
    "file-loader": "^1.1.11",
    "fiori-fundamentals": "1.5.3",
    "fontfaceobserver": "^2.0.13",
    "handlebars": "^4.7.3",
    "jest": "^24.0.0",
    "jest-styled-components": "^4.11.0-1",
    "npm-run-all": "^4.1.1",
    "raf": "^3.4.0",
    "react-jsonschema-form": "1.7.0",
    "react-modal": "^3.5.1",
    "react-scrollbar": "^0.5.4",
    "react-test-renderer": "^16.0.0",
    "style-loader": "^0.20.2",
    "styled-components": "^3.2.3",
    "styled-components-grid": "^2.2.0",
    "webpack": "^4.28.1",
    "webpack-cli": "^3.2.1",
    "webpack-merge": "^4.2.1"
  },
  "peerDependencies": {
    "styled-components": "^3.2.3"
  },
  "resolutions": {
    "uglify-es": "3.3.7"
  },
  "jest": {
    "collectCoverage": true,
    "collectCoverageFrom": [
      "src/**",
      "!src/**/*.snap",
      "!src/components/index.js",
      "!src/modifiers/**",
      "!src/index.js"
    ],
    "moduleNameMapper": {
      "^.+\\.(css|less|scss)$": "babel-jest"
    },
    "moduleDirectories": [
      "node_modules",
      "./"
    ],
    "setupFiles": [
      "raf/polyfill",
      "./test_setup/config"
    ]
  }
}
