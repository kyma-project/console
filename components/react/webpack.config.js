const webpack = require('webpack');
const path = require('path');
const pkg = require('./package.json');

require('babel-polyfill');

let libraryName = pkg.name;

module.exports = {
  entry: ['./src/index'],
  module: {
    loaders: [
      { test: /\.js?$/, loader: 'babel-loader', exclude: /node_modules/ },
      {
        test: /\.css$/,
        use: [{ loader: 'style-loader' }, { loader: 'css-loader' }],
      },
      {
        test: /\.(eot|svg|ttf|woff|woff2|otf)$/,
        loader: 'file-loader?name=fonts/[name].[ext]',
      },
    ],
  },
  resolve: {
    modules: [path.resolve('./node_modules'), 'node_modules'],
    extensions: ['.js'],
  },
  output: {
    path: path.join(__dirname, '/lib'),
    publicPath: '/',
    filename: 'index.js',
    library: libraryName,
    libraryTarget: 'umd',
    umdNamedDefine: true,
  },
  devtool: 'source-map',
  devServer: {
    contentBase: './src',
    hot: true,
  },
  plugins: [
    new webpack.optimize.OccurrenceOrderPlugin(),
    new webpack.HotModuleReplacementPlugin(),
    new webpack.NoEmitOnErrorsPlugin(),
  ],
  externals: {
    'styled-components': {
      commonjs: 'styled-components',
      commonjs2: 'styled-components',
      amd: 'styled-components',
    },
  },
};
