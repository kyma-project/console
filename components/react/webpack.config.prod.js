const webpack = require('webpack');
const path = require('path');
const pkg = require('./package.json');
const webpackConfigBase = require('./webpack.config');
require('babel-polyfill');

const plugins = Array.from(webpackConfigBase.plugins);

plugins.push(
  new webpack.DefinePlugin({
    'process.env.NODE_ENV': JSON.stringify('production'),
  }),
);

const webpackConfig = {
  ...webpackConfigBase,
  devtool: false,
  plugins,
};

module.exports = webpackConfig;
