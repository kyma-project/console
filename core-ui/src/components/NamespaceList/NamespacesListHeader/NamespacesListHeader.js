import React from 'react';
import PropTypes from 'prop-types';
import { Panel, Search } from '@kyma-project/react-components';
import LuigiClient from '@kyma-project/luigi-client';

import NamespaceFilters from './NamespaceFilters/NamespaceFilters';
import './NamespacesListHeader.scss';
import ModalWithForm from '../../ModalWithForm/ModalWithForm';
import CreateNamespaceForm from '../../CreateNamespaceForm/CreateNamespaceForm';

NamespacesListHeader.propTypes = {
  labelFilters: PropTypes.arrayOf(PropTypes.object.isRequired).isRequired,
  updateSearchPhrase: PropTypes.func.isRequired,
  setLabelFilters: PropTypes.func.isRequired,
};

function navigateToNamespaceDetails(namespaceName) {
  LuigiClient.linkManager().navigate(
    `/home/namespaces/${namespaceName}/details`,
  );
}

export default function NamespacesListHeader({
  labelFilters,
  updateSearchPhrase,
  setLabelFilters,
}) {
  const queryParameters = LuigiClient.getNodeParams();
  const opened = queryParameters && queryParameters.showModal === 'true';

  const clearQueryParams = () => {
    LuigiClient.linkManager().navigate('/');
  };

  return (
    <Panel className="namespace-list-header fd-has-padding-medium remove-after">
      <span className="fd-has-type-4">Namespaces</span>
      <div className="namespace-list-actions">
        <Search
          onChange={e => updateSearchPhrase(e.target.value)}
          data-test-id="namespace-seach-input"
          className="search--white"
        />
        <NamespaceFilters
          filters={labelFilters}
          updateFilters={setLabelFilters}
        />
        <ModalWithForm
          title="Add new namespace"
          button={{ text: 'Add new namespace', glyph: 'add' }}
          id="add-namespace-modal"
          renderForm={props => (
            <CreateNamespaceForm
              {...props}
              onCompleted={navigateToNamespaceDetails}
            />
          )}
          opened={opened}
          customCloseAction={clearQueryParams}
        />
      </div>
    </Panel>
  );
}
