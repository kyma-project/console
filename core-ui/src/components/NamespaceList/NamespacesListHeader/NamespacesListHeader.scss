.namespace-list-header {
  @media (min-width: 320px) {
    display: flex;
    justify-content: space-between;
  }

  .namespace-list-actions > * {
    margin-left: 16px;
    display: inline-flex;
  }
}

.remove-after::after {
  content: unset !important;
}

// TODO Delete after upgrading search
.search--white.fd-search-input {
  .fd-input {
    color: var(--fd-color-text-1);
    background-color: var(--fd-forms-background-color);
    border-color: var(--fd-forms-border-color);
  }
  .fd-input-group__addon{
    border-color: var(--fd-forms-border-color);
    background-color: var(--fd-color-neutral-1);
  }
}

#add-namespace-modal {
  .fd-modal {
    max-width: 90%;
  }
  .fd-modal__content {
    width: 50em;
    min-width: 460px;
  }
}
