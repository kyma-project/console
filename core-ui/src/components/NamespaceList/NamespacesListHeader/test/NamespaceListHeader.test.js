import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import NamespacesListHeader from '../NamespacesListHeader';

jest.mock('@kyma-project/luigi-client', () => ({
  getNodeParams: () => ({
    showModal: 'false',
  }),
  uxManager: () => ({
    addBackdrop: () => {},
    removeBackdrop: () => {},
  }),
  linkManager: () => ({
    navigate: () => {},
  }),
}));

describe('NamespacesListHeader', () => {
  it('Renders with minimal props', () => {
    const { queryByLabelText } = render(
      <NamespacesListHeader
        labelFilters={[]}
        updateSearchPhrase={() => {}}
        setLabelFilters={() => {}}
      />,
    );

    const title = queryByLabelText('title');
    expect(title).toBeInTheDocument();
    expect(title).toHaveTextContent('Namespaces');
  });

  it('Fires callback on input field change', () => {
    const mockCallback = jest.fn();

    const { getByPlaceholderText } = render(
      <NamespacesListHeader
        labelFilters={[{ name: '1' }, { name: '2' }]}
        updateSearchPhrase={mockCallback}
        setLabelFilters={() => {}}
      />,
    );

    const input = getByPlaceholderText('Search...');
    fireEvent.change(input, { target: { value: 'a' } });

    expect(mockCallback).toHaveBeenCalledTimes(1);
    expect(mockCallback).toHaveBeenCalledWith('a');
  });
});
