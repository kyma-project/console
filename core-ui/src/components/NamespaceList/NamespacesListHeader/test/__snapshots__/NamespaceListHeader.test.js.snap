// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NamespacesListHeader UI Renders with minimal props 1`] = `
<div
  className="fd-panel namespace-list-header fd-has-padding-medium remove-after"
>
  <span
    className="fd-has-type-4"
  >
    Namespaces
  </span>
  <div
    className="namespace-list-actions"
  >
    <div
      className="fd-search-input search--white"
      data-test-id="namespace-seach-input"
    >
      <div
        className="fd-popover"
      >
        <div
          className="fd-popover__control"
        >
          <div
            aria-expanded={false}
            aria-haspopup="true"
            className="fd-combobox-control"
          >
            <div
              className="fd-input-group fd-input-group--after"
            >
              <input
                className="fd-input"
                onChange={[Function]}
                onClick={[Function]}
                onKeyPress={[Function]}
                style={Object {}}
                type="text"
                value=""
              />
              <span
                className="fd-input-group__addon fd-input-group__addon--after fd-input-group__addon--button"
              >
                <button
                  className="fd-button--light sap-icon--search"
                  onClick={[Function]}
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="fd-popover"
    >
      <div
        onFocus={[Function]}
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      >
        <div
          className="fd-popover__control"
        >
          <button
            className="fd-button--light sap-icon--filter"
            onClick={[Function]}
          />
        </div>
      </div>
    </div>
    <div>
      <button
        className="fd-button sap-icon--add"
        onClick={[Function]}
      >
        Add new namespace
      </button>
    </div>
  </div>
</div>
`;
